src/Cardano/Prelude/GHC/Heap/Size.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE MagicHash           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UnboxedTuples       #-}
+
+module Cardano.Prelude.GHC.Heap.Size (
+    computeHeapSize
+  ) where
+
+import Cardano.Prelude.Base hiding (Any)
+
+import GHC.Exts
+import GHC.Exts.Heap
+import System.Mem (performGC)
+
+computeHeapSize :: a -> IO Word64
+computeHeapSize = \a -> do
+    performGC
```

Comment:
Let's document why we do a GC first, and add a comment saying we _could_ end up counting stuff twice (if things get moved during GC).